p2p/src/lib.rs

```diff
@@ -18,44 +19,68 @@ pub mod peer;
 /// The main type to use for secure communication.
 pub type Network<T> = NetworkBase<T, X25519Sha256, ChaCha20Poly1305>;
 
-/// Error types of this crate.
-#[derive(Debug, FromVariant, Error, iroha_actor::Message)]
+/// Errors used in the [`iroha_p2p`] crate.
+#[derive(Clone, Debug, iroha_derive::FromVariant, Error, iroha_actor::Message)]
 pub enum Error {
     /// Failed to read or write
-    #[error("Failed IO operation")]
-    Io(#[source] io::Error),
+    #[error("Failed IO operation. {0}")]
+    Io(#[source] std::sync::Arc<io::Error>),
     /// Failed to read or write
-    #[error("Failed handshake")]
-    Handshake,
+    #[error("{0}: Failed handshake")]
+    Handshake(u32),
     /// Failed to read or write
-    #[error("Failed reading message")]
+    #[error("Message improperly formatted")]
     Format,
     /// Failed to create keys
     #[error("Failed to create session key")]
     Keys,
     /// Failed to parse address
-    #[error("Failed to parse socket address")]
+    #[error("Failed to parse socket address. {0}")]
```

Comment:
You shouldnt pring source error here, as it will be printed as source

Reply:
It should be, but specifying source doesn't always work: [see this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=346bce4c5ddfe2278b227741bae5d85b)

Reply:
It shouldnt be printed though in display. Display is a way to print exactly this level of error without its source. [See this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=852fefb5fb97435e8d2c735d3e7483eb)